common/src/main/scala/org/apache/celeborn/common/meta/PartitionLocationInfo.scala

```diff
@@ -201,6 +234,24 @@ class PartitionLocationInfo extends Logging {
     }
   }
 
+  def removeAllRelatedPartitions(
+      shuffleKey: String,
+      partitionIdOpt: Option[Int]): Unit = this
+    .synchronized {
+      partitionIdOpt match {
+        case Some(partitionId) =>
+          if (masterPartitionLocations.containsKey(shuffleKey)) {
+            masterPartitionLocations.get(shuffleKey).remove(partitionId)
```

Comment:
Seems the logic to remove mappartition is inconsistent with reducepartition

Reply:
lifecycleManager need release only its memory as controller wants release its slots but they use the same function, (Am I right?) If so, It's can be improved later.

Reply:
the release slots is mainly for Controller, agree to improve later